### ISSUE TITLE
Evaluate usage of materialized views for the scoreboard

### ISSUE THREAD

**Issue by Trolldemorted** [enowars/EnoEngine#55]
Faust uses them: https://github.com/fausecteam/ctf-gameserver/blob/1ae31b406faad89719ca93e7a0f273a25895c153/doc/controller/scoring.sql

They might be way faster than our current evaluation.

**Comment by domenukk** [comment]
Is this really necessary?
It feels hard to understand, tweak, and maintain.
The bus factor is already low for the engine, but this reduces it to < 1.

**Comment by Trolldemorted** [comment]
That depends on how long the round time calculation is allowed to take for a large number of teams and submissions, obviously 😛

Do you think faust's code looks so bad?

**Comment by domenukk** [comment]
I heard not everybody at Faust knows what's going on there.
The round time calculation should not be longer than the round time...
What's the current perf?

**Comment by domenukk** [comment]
The calculations could be done doing some mocking, right? Didn't Moritz write something last year?

**Comment by Trolldemorted** [comment]
Yeah, but I still need a system with a decent ssd and some memory for a fair test

**Comment by ldruschk** [comment]
The OpenNebula host has a Xeon E5-2609 v4, 32 GB RAM but unfortunately only an HDD. But since MySQL should cache most of the stuff in RAM this should still be somewhat usable, right?

**Comment by Trolldemorted** [comment]
our sliding windows over the last n rounds should be good enough, further optimization will be more agressive parallelization
